packages/calypso-e2e/src/jest-playwright-config/environment.ts

```diff
@@ -373,6 +373,18 @@ function setupBrowserProxyTrap( browser: Browser ): Browser {
 					// Default value is 15000ms, defined in env-variables.ts.
 					page.setDefaultTimeout( env.TIMEOUT );
 
+					// Set up a HTTP response status interceptor
+					// to capture instances of 502 Bad Gateway.
+					// This remains active until the page is
+					// closed.
+					page.on( 'response', ( response ) => {
+						if ( response.status() === 502 ) {
+							throw new Error(
+								`Encountered HTTP ${ response.status } on request to URL ${ response.url() } }`
```

Comment:
@WunderBart or @noahtallen - typically when a 502 is encountered, would a better course of action be to refresh the page instead? 

The one time I encountered 502 on production while using Calypso, refreshing loaded the correct page, but I'm a bit torn on how to handle it on our E2Es.

Reply:
I think a simple reload on `502` could be beneficial as we wouldn't need to restart the whole job. Would the page object still be connected after a reload via this context, though? I'm also unsure if this wouldn't obfuscate a potentially repeating 502s - or is this covered by tracking, @noahtallen?

Reply:
We can try reloading on a 502 once by using `page.once` instead of `page.on`.

Reply:
Good question... Every single production request is tracked in logstash, so we can see any type of Calypso server error.

That said, 502 bad gateway probably doesn't even come from the Calypso server. It probably means nginx can't connect to the calypso server for whatever reason. 🤔 That's just a guess, but it wouldn't show up in the calypso server logs directly. 

But _that said_, we do have logging for slow response times, which could be related. Every single one is from `/cspreport` -- iirc that was also causing problems in e2e? Is that related to this fix?

I think a refresh makes sense as well. That's what a user would probably do, and if I had to guess, an e2e test isn't going to detect the root cause of a 502 error anyways. In other words, if there was a severe server issue (like it completely crashed), then the entire suites would permafail. If there is just an intermittent issue, we can't really use e2es to find that issue anyways. So I think adding a refresh for less-flaky e2es would be helpful.